one/src/migrations.rs

```diff
@@ -0,0 +1,157 @@
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result};
+use async_stream::try_stream;
+use async_trait::async_trait;
+use ceramic_event::unvalidated;
+use ceramic_metrics::{config::Config as MetricsConfig, init_local_tracing};
+use ceramic_service::{Block, BoxedBlock};
+use cid::Cid;
+use clap::{Args, Subcommand};
+use futures::Stream;
+use multihash_codetable::{Code, Multihash, MultihashDigest};
+use tracing::{debug, info};
+
+use crate::{DBOpts, Info};
+
+#[derive(Subcommand, Debug)]
+pub enum EventsCommand {
+    /// Migrate raw event blocks from IPFS.
+    FromIpfs(FromIpfsOpts),
+}
+
+#[derive(Args, Debug)]
+pub struct FromIpfsOpts {
+    /// The path to the ipfs_repo [eg: ~/.ipfs/blocks]
+    #[clap(long, short, value_parser)]
+    input_ipfs_path: PathBuf,
+
+    /// The database to connect to e.g.
+    ///
+    /// `sqlite:///path/to/file.sqlite3` or `sqlite://:memory:` or `sqlite://~/.ceramic-one/db.sqlite3`
+    #[clap(long, short, default_value = "sqlite://~/.ceramic-one/db.sqlite3")]
```

Comment:
should we make a constant to ensure that the default used here is the same as the default used for the daemon?

Reply:
This is unfortunately not the same concept. The daemon has a combination of the store path and a db url that define where to find the db. The migration cmd doesn't have a concept of a store path. So it has a sane default.

We should figure how to reconcile these differences as they are confusing.

Reply:
@dav1do Any ideas on how to simplify this?

Reply:
We could probably remove the `--database-url` and rely on the store directory exclusively as we have no plans to support postgres or alternative databases. It did use it for some testing yesterday because I had custom database names, but we probably don't want to allow people to rename the file at this point. And it's kind of bugged because if you pass a database-url it might not be in your store-dir at all, and then it's just weird (you could even pass `sqlite//:memory:` if you were so inclined).

For now, we should probably just no-op it and print a warning. I was just noodling about removing the store crate from ceramic-one and going through the service crate for everything (like give me a service using sqlite or something). 

I'm not sure how/if we want to specify database files/types, but I think we should probably say the file name/format is up to us and you can only provide a directory. Not sure how likely it is, but I could imagine creating multiple sqlite files for like every ~100gb of events or something. Probably not how we'd want to shard things, but the point is mostly that it's not really an option if we let people pick the sqlite file name and expect only one.

Reply:
Maybe I kind of missed the point in that response 😅 but the daemon has  database-url to specify and it will not be modified or moved from that location, or it will be called `db.sqlite3` in one of these 3 places:

1. path from options
2. path $HOME/.ceramic-one
3. pwd/.ceramic-one

IMO specifying is something we don't want (even though I'm the one that added it 🤦).

Reply:
https://github.com/ceramicnetwork/rust-ceramic/pull/384